staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -929,11 +930,35 @@ func (r *Request) newHTTPRequest(ctx context.Context) (*http.Request, error) {
 	if err != nil {
 		return nil, err
 	}
-	req = req.WithContext(ctx)
+	req = req.WithContext(httptrace.WithClientTrace(ctx, newDNSMetricsTrace(ctx)))
```

Comment:
Since you're touching this, do you want to fold this and the above into a single call to `http.NewRequestWithContext`? That would avoid a needless shallow copy of every Request.

Reply:
Done